data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3ObjectWorker.java

```diff
@@ -99,21 +119,35 @@ void parseS3Object(final S3ObjectReference s3ObjectReference) throws IOException
                 .build();
 
         final BufferAccumulator<Record<Event>> bufferAccumulator = BufferAccumulator.create(buffer, numberOfRecordsToAccumulate, bufferTimeout);
-        try {
-            s3ObjectReadTimer.recordCallable((Callable<Void>) () -> {
-                doParseObject(s3ObjectReference, getObjectRequest, bufferAccumulator);
-
-                return null;
-            });
-        } catch (final IOException | RuntimeException e) {
-            throw e;
-        } catch (final Exception e) {
-            // doParseObject does not throw Exception, only IOException or RuntimeException. But, Callable has Exception as a checked
-            // exception on the interface. This catch block thus should not be reached, but in case it is, wrap it.
-            throw new RuntimeException(e);
-        }
-
-        s3ObjectsSucceededCounter.increment();
+        //Issue#1971-Start
+		if (null != s3SourceConfig.getS3SelectOptions()) {
+			try {
+				s3SelectReadTimer.recordCallable((Callable<Void>) () -> {
+					s3SelectObjectWorker.selectObjectFromS3(s3ObjectReference, bufferAccumulator);
+					s3SelectRecordsSucceeded.increment();
+					return null;
+				});
+			} catch (final IOException | RuntimeException e) {
+				throw e;
+			} catch (final Exception e) {
+				throw new RuntimeException(e);
+			}
+		} else {
```

Comment:
I see a lot of formatting which is off. Please use 4 spaces and no tabs.

Reply:
Thanks, David for your review. we have incorporated the comment and will raise pull request for your review.